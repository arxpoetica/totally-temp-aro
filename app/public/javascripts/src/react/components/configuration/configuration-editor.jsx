import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import reduxStore from '../../../redux-store'
import wrapComponentWithProvider from '../../common/provider-wrapped-component'
import ConfigurationActions from '../configuration/configuration-actions'
import SettingsEditor from './settings-editor.jsx'

export class ConfigurationEditor extends Component {
  constructor (props) {
    super(props)
    // Form component state does not need to go into Redux
    this.state = {
      showInitialWarning: true
    }
  }

  render () {
    return <div>
      { this.state.showInitialWarning
        ? this.renderWarningMessage()
        : <SettingsEditor id='compSettingsEditor' />
      }
    </div>
  }

  renderWarningMessage () {
    return <div id='divConfigurationWarning'>
      <div className='alert alert-danger'>
        <strong>Careful:</strong> These settings are intended to be changed by system administrators. Incorrect settings
        can cause the entire platform to stop working <strong>for all users</strong>.
      </div>
      <button id='btnAcceptConfigurationWarning' className='btn btn-block btn-danger'
        onClick={event => this.setState({ showInitialWarning: false })}>
        I understand, take me to the configuration editor
      </button>
    </div>
  }
}

ConfigurationEditor.propTypes = {
  configuration: PropTypes.object
}

const mapStateToProps = (state) => ({
  initialConfiguration: state.configuration
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  saveConfigurationToServerAndReload: (type, configuration) => dispatch(ConfigurationActions.saveConfigurationToServerAndReload(type, configuration))
})

const ConfigurationEditorComponent = wrapComponentWithProvider(reduxStore, ConfigurationEditor, mapStateToProps, mapDispatchToProps)
export default ConfigurationEditorComponent
