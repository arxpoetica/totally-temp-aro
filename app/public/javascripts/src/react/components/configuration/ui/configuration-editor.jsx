import React, { Component } from 'react'
import reduxStore from '../../../../redux-store'
import wrapComponentWithProvider from '../../../common/provider-wrapped-component'
import UiActions from './ui-actions'
import SettingsEditor from './settings-editor.jsx'
import AssetManager from './asset-manager.jsx'
import StylesheetsManager from './stylesheets-manager.jsx'
import EtlTemplateManager from './etl-template-manager.jsx'

export class ConfigurationEditor extends Component {
  constructor (props) {
    super(props)
    this.navs = Object.freeze({
      SETTINGS: 'Settings',
      ASSET_MANAGER: 'Asset Manager',
      STYLESHEETS: 'Stylesheets',
      ETL_TEMPLATE_MANAGER: 'Templates'
    })
    this.state = {
      showInitialWarning: true,
      selectedNav: this.navs.SETTINGS
    }
  }
  navSelection () {
    let val = this.state.selectedNav

    if (val === 'Settings') {
      return (
        <SettingsEditor id='compSettingsEditor' />
      )
    } else if (val === 'Asset Manager') {
      return (
        <AssetManager />
      )
    } else if (val === 'Stylesheets') {
      return (
        <StylesheetsManager />
      )
    } else {
      return (
        <EtlTemplateManager />
      )
    }
  }
  render () {
    return <div>
      { this.state.showInitialWarning
        ? this.renderWarningMessage()
        : this.renderControls()
      }
    </div>
  }

  renderWarningMessage () {
    return <div id='divConfigurationWarning'>
      <div className='alert alert-danger'>
        <strong>Careful:</strong> These settings are intended to be changed by system administrators. Incorrect settings
        can cause the entire platform to stop working <strong>for all users</strong>.
      </div>
      <button id='btnAcceptConfigurationWarning' className='btn btn-block btn-danger'
        onClick={event => this.setState({ showInitialWarning: false })}>
        I understand, take me to the configuration editor
      </button>
    </div>
  }

  renderControls () {
    return <div>
      <ul className='nav nav-tabs' style={{ marginBottom: '10px' }}>
        <li className='nav-item' onClick={() => this.setState({ selectedNav: this.navs.SETTINGS })}>
          <a className={`nav-link ${this.state.selectedNav === this.navs.SETTINGS ? 'active' : ''}`} href='#'>
            {this.navs.SETTINGS}
          </a>
        </li>
        <li className='nav-item' onClick={() => this.setState({ selectedNav: this.navs.ASSET_MANAGER })}>
          <a className={`nav-link ${this.state.selectedNav === this.navs.ASSET_MANAGER ? 'active' : ''}`} href='#'>
            {this.navs.ASSET_MANAGER}
          </a>
        </li>
        <li className='nav-item' onClick={() => this.setState({ selectedNav: this.navs.STYLESHEETS })}>
          <a className={`nav-link ${this.state.selectedNav === this.navs.STYLESHEETS ? 'active' : ''}`} href='#'>
            {this.navs.STYLESHEETS}
          </a>
        </li>
        <li className='nav-item' onClick={() => this.setState({ selectedNav: this.navs.ETL_TEMPLATES })}>
          <a className={`nav-link ${this.state.selectedNav === this.navs.ETL_TEMPLATES ? 'active' : ''}`} href='#'>
            {this.navs.ETL_TEMPLATE_MANAGER}
          </a>
        </li>
      </ul>
      {
        this.navSelection()
        // (this.state.selectedNav === this.navs.SETTINGS)
        //   ? <SettingsEditor id='compSettingsEditor' />
        //   : <AssetManager />
      }
    </div>
  }
}

ConfigurationEditor.propTypes = {}

const mapStateToProps = (state) => ({})

const mapDispatchToProps = (dispatch, ownProps) => ({
  saveConfigurationToServerAndReload: (type, configuration) => dispatch(UiActions.saveConfigurationToServerAndReload(type, configuration))
})

const ConfigurationEditorComponent = wrapComponentWithProvider(reduxStore, ConfigurationEditor, mapStateToProps, mapDispatchToProps)
export default ConfigurationEditorComponent
