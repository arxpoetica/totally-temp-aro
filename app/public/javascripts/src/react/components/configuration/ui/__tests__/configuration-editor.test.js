/* global test expect */
import React from 'react'
import { shallow } from 'enzyme'
import { ConfigurationEditor } from '../configuration-editor'

const sampleConfiguration = {
  locationCategories: {
    medium_business: {
      label: 'Medium businesses'
    }
  }
}

// -----------------------------------------------------------------------------
test('Component render - default and after clicking button', () => {
  const component = shallow(
    <ConfigurationEditor initialConfiguration={sampleConfiguration} />
  )
  // Initial state - show warning, hide form
  expect(component).toMatchSnapshot()
  expect(component.find('#divConfigurationWarning').exists()).toBeTruthy()
  expect(component.find('#compSettingsEditor').exists()).toBeFalsy()
  component.find('#btnAcceptConfigurationWarning').simulate('click')

  // New state - hide warning, show form
  expect(component.find('#divConfigurationWarning').exists()).toBeFalsy()
  expect(component.find('#compSettingsEditor').exists()).toBeTruthy()
  //expect(component).toMatchSnapshot()
})
