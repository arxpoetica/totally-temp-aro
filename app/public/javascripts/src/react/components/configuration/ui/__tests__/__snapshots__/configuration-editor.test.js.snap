// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render - default and after clicking button 1`] = `
<div>
  <div
    id="divConfigurationWarning"
  >
    <div
      className="alert alert-danger"
    >
      <strong>
        Careful:
      </strong>
       These settings are intended to be changed by system administrators. Incorrect settings can cause the entire platform to stop working 
      <strong>
        for all users
      </strong>
      .
    </div>
    <button
      className="btn btn-block btn-danger"
      id="btnAcceptConfigurationWarning"
      onClick={[Function]}
    >
      I understand, take me to the configuration editor
    </button>
  </div>
</div>
`;

exports[`Component render - default and after clicking button 2`] = `
<div>
  <div>
    <ul
      className="nav nav-tabs"
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      <li
        className="nav-item"
        onClick={[Function]}
      >
        <a
          className="nav-link active"
          href="#"
        >
          Settings
        </a>
      </li>
      <li
        className="nav-item"
        onClick={[Function]}
      >
        <a
          className="nav-link "
          href="#"
        >
          Asset Manager
        </a>
      </li>
      <li
        className="nav-item"
        onClick={[Function]}
      >
        <a
          className="nav-link "
          href="#"
        >
          Stylesheets
        </a>
      </li>
    </ul>
    <Component
      id="compSettingsEditor"
    />
  </div>
</div>
`;
